packages/fetchai/connections/http/connection.py

```diff
@@ -0,0 +1,384 @@
+
+        if kwargs.get("connection_id") is None:
+            kwargs["connection_id"] = PublicId("fetchai", "http", "0.1.0")
+
+        super().__init__(*args, **kwargs)
+        self.address = address
+        self.channel = HTTPChannel(
+            address,
+            host,
+            port,
+            api_spec,
+            connection_id=self.connection_id,
+            excluded_protocols=self.excluded_protocols,
+        )
+
+    async def connect(self) -> None:
+        """
+        Connect to the http.
+
+        :return: None
+        """
+        if not self.connection_status.is_connected:
+            self.connection_status.is_connected = True
+            self.channel.in_queue = asyncio.Queue()
+            self.channel.loop = self.loop
+            self.channel.connect()
+
+    async def disconnect(self) -> None:
+        """
+        Disconnect from HTTP.
+
+        :return: None
+        """
+        if self.connection_status.is_connected:
+            self.connection_status.is_connected = False
+            self.channel.disconnect()
+
+    async def send(self, envelope: "Envelope") -> None:
+        """
+        Send an envelope.
+
+        :param envelope: the envelop
+        :return: None
+        """
+        if not self.connection_status.is_connected:
+            raise ConnectionError(
+                "Connection not established yet. Please use 'connect()'."
+            )  # pragma: no cover
+        # self.channel.send(envelope)
+
+    async def receive(self, *args, **kwargs) -> Optional["Envelope"]:
+        """
+        Receive an envelope.
+
+        :return: the envelope received, or None.
+        """
+        if not self.connection_status.is_connected:
+            raise ConnectionError(
+                "Connection not established yet. Please use 'connect()'."
+            )  # pragma: no cover
+        assert self.channel.in_queue is not None
+        try:
+            envelope = await self.channel.in_queue.get()
+            if envelope is None:
+                return None  # pragma: no cover
+
+            return envelope
+        except CancelledError:  # pragma: no cover
+            return None
+
+    @classmethod
+    def from_config(
+        cls, address: Address, connection_configuration: ConnectionConfig
+    ) -> "Connection":
+        """
+        Get the HTTP connection from the connection configuration.
+
+        :param address: the address of the agent.
+        :param connection_configuration: the connection configuration object.
+            :host - RESTful API hostname / IP address
+            :port - RESTful API port number
+            :api - Directory API path and filename of the API spec YAML source file.
+        :return: the connection object
+        """
+        host = cast(str, connection_configuration.config.get("host"))
+        port = cast(int, connection_configuration.config.get("port"))
+        api = cast(str, connection_configuration.config.get("api")) if not "" else None
```

Comment:
should be
```suggestion
        api_specification_path = cast(str, connection_configuration.config.get("api_specification_path"), "")
```

Reply:
the `get` lets you specify a default value

Reply:
fixed as `api_spec_path = cast(str, connection_configuration.config.get("api_spec_path"), "")`, will be there next commit.

Reply:
Update, revert back to `api_spec_path = cast(str, connection_configuration.config.get("api_spec_path"))` as mypy has been given me the following errors:

```
connection.py:400: error: 'cast' expects 2 arguments
connection.py:400: error: No overload variant of "cast" matches argument types "Any", "Any", "str"
connection.py:400: note: Possible overload variants:
connection.py:400: note:     def [_T] cast(tp: Type[_T], obj: Any) -> _T
connection.py:400: note:     def cast(tp: str, obj: Any) -> Any
```